spec/models/manageiq/providers/openstack/infra_manager/host_spec.rb

```diff
@@ -515,5 +519,17 @@
       expect(task.status).to eq("Ok")
       expect(MiqQueue.where(:method_name => "refresh").count).to eq(1)
     end
+
+    it "check task executes and queues destroy_queue if success" do
+      allow_any_instance_of(ManageIQ::Providers::Openstack::InfraManager).to receive(:openstack_handle).and_return([])
+      allow_any_instance_of(Array).to receive(:detect_baremetal_service).and_return([])
```

Comment:
`Array` seems strange here ... seems like it would be easy enough to use a `double` to receive these messages.

I know there are some [complicated reasons](https://github.com/rspec/rspec-mocks/blob/4d842fefb82eaca7d33554746c1fb8c03738b068/lib/rspec/mocks/any_instance/recorder.rb#L197-L218) to avoid `allow_any_instance_of`.  So, if you can `double` this away, it would be better.